### DISCUSSION TITLE
nova-2 model started returning empty strings as transcription results

### DISCUSSION THREAD

**Discussion by heikir** [deepgram/community#826] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Streaming

### Details

I am sending 8khz mulaw audio stream from Twilio to Deepgram and since a few days, the API started returning empty strings in the results of the API. Logging `data.channel.alternatives` gives me the result `[ { transcript: '', confidence: 0, words: [] } ]` even though when saving the recorded audio into a local .wav file, the recording works and also matches the audio length that Deepgram receives. I am not sure what changed for the API, but the same code used to work a week ago.

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

/v1/listen?model=nova-2-phonecall&encoding=mulaw&sample_rate=8000&language=en-US&smart_format=true&channels=1

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

64984aab-24a3-4600-af93-acbc2b333234

### If possible, please attach an example audio file to reproduce the issue.

[Voicebot Demo June 21.wav.zip](https://github.com/user-attachments/files/15933928/Voicebot.Demo.June.21.wav.zip)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hey @heikir, this means the audio is not in the format defined by `encoding=mulaw&sample_rate=8000&channels=1`. If you can get a raw dump of the binary data, you can run `ffmpeg -ar 8k -ac 1 -f mulaw -i <file_containing_binary_data> out.wav` to convert the file to a WAV file, given a sample rate of 8k, one channel, and mulaw encoding.

Then you can listen to the file to verify if the audio is garbled or clear.

**Comment by heikir** [comment]
@jjmaldonis I am already saving the audio as a .wav on disk as well to check whether it's good quality and the recordings are clear. Also Twilio is very clear that the format they send the data is 8 bit mulaw at 8k sample rate: https://www.twilio.com/docs/voice/media-streams/websocket-messages#start-message

Is there any way for me to retrieve the files I send to Deepgram to check what your service receives?

**Reply by KADURKA-ANIL** [reply]
@heikir bro did u fix the issue like i am also facing the same situation but upon getting stream from twillio i am decoding suing base64 that converts the payload we get from twilio into the class bytes ,now i am sending but we are getting the empty string .i have  tried with the 2 differnet repos with two different approaches but i am getting same empty stirng .so if you found any improvement from your side .

**Reply by jjmaldonis** [reply, marked as answer]
Hey @heikir, we don't provide a way to download the audio you've sent to Deepgram, but I'm able to review the file for debugging. Noticed the binary data ends with `==` which implies the data is base64 encoded. When I tried decoding the binary data you sent using a base64 decoder, the decoding succeeded which means the file is valid base64 encoded text.

I'm guessing the Twilio setup is sending the data as base64 encoded data, and Deepgram requires raw binary data to be sent over the websocket.

When I ran the decoded binary data through Deepgram's pre-recorded API using the same parameters you provided, the transcript still came back empty. So there may be something else going on as well -- but the data definitely needs to be base64 decoded before sending it to us.

**Reply by heikir** [reply]
I found my mistake - I was turning the base64 into raw bytes and turning them back into a string again because node.js streams work easiest with string-format data. It would be very helpful to give me access to the raw data that I send to deepgram for debugging purposes so that I could have run that analysis myself. Thanks for your support here though! 🙏🏼

Closing the ticket as resolved.

**Comment by heikir** [comment]
Issue resolved through debugging the sent data file format.
